### PR TITLE
Update Staking and DAO according to audit feedbacks

### PR DESCRIPTION
**Pull request by manhlx3006**
Changes:
1. Using struct for staker's latest data + staker's data per epoch in KyberStaking, campaign and vote data in KyberDAO
2. Remove code duplication for get latest network or brr data
3. Fix naming convention and style

Can check each commit changes

### CODE REVIEW

File: contractsSol5/Dao/KyberDAO.sol [hunk 1]
```diff
@@ -470,81 +439,75 @@ contract KyberDAO is IKyberDAO, EpochUtils, ReentrancyGuard, CampPermissionGroup
...
     }
 
     /**
     * @dev return campaign winning option and its value
-    * @dev return (0, 0) if campaign does not existed
-    * @dev return (0, 0) if campaign is not ended yet
-    * @dev return (0, 0) if campaign has no winning option based on the formula
-    * @param campID id of campaign to get result
+    *      return (0, 0) if campaign does not existed
+    *      return (0, 0) if campaign is not ended yet
```
Comment (ilanDoron) [inline contractsSol5/Dao/KyberDAO.sol:471]: typos:
existed --> exist
is not ended --> has not ended.

File: contractsSol5/Dao/mock/MockMaliciousDAO.sol [hunk 2]
```diff
@@ -7,17 +7,15 @@ contract MockMaliciousDAO is KyberDAO {
...
         address _admin
     ) KyberDAO(
         _epochPeriod, _startTimestamp,
         _staking, _feeHandler, _knc,
         _defaultNetworkFeeBps, _defaultRewardBps, _defaultRebateBps,
         _admin
     ) public {
-        MAX_CAMP_OPTIONS = _maxNumOptions;
-        MIN_CAMP_DURATION_SECONDS = _minCampDuration;
+        MIN_CAMPAIGN_DURATION_SECONDS = _minCampDuration;
```
Comment (ilanDoron) [inline contractsSol5/Dao/mock/MockMaliciousDAO.sol:18]: camp --> campaign
MIN_CAMPAIGN_DURATION_SECONDS if not const. should be camel cased

File: contractsSol5/IKyberDAO.sol [hunk 3]
```diff
@@ -5,11 +5,12 @@ interface IKyberDAO {
     function EPOCH_PERIOD_SECONDS() external view returns(uint);
     function FIRST_EPOCH_START_TIMESTAMP() external view returns(uint);
 
+    function vote(uint campID, uint option) external;
```
Comment (ilanDoron) [inline contractsSol5/IKyberDAO.sol:8]: camp --> campaign

File: contractsSol5/IKyberDAO.sol [hunk 4]
```diff
@@ -5,11 +5,12 @@ interface IKyberDAO {
     function EPOCH_PERIOD_SECONDS() external view returns(uint);
```
Comment (ilanDoron) [inline contractsSol5/IKyberDAO.sol:5]: naming convention for these functions.

File: contractsSol5/Dao/KyberDAO.sol [hunk 5]
```diff
@@ -77,66 +76,73 @@ contract CampPermissionGroups {
...
-    // max number of camps for each epoch
-    uint public constant MAX_EPOCH_CAMPS = 10;
+    // max number of campaigns for each epoch
+    uint public constant MAX_EPOCH_CAMPAIGNS = 10;
     // max number of options for each campaign
-    uint public MAX_CAMP_OPTIONS = 8;
+    uint public constant MAX_CAMPAIGN_OPTIONS = 8;
     // minimum duration in seconds for a campaign
-    uint public MIN_CAMP_DURATION_SECONDS = 345600; // around 4 days
+    uint public MIN_CAMPAIGN_DURATION_SECONDS = 345600; // around 4 days
```
Comment (ilanDoron) [inline contractsSol5/Dao/KyberDAO.sol:84]: move to non constant naming convention.

File: contractsSol5/Dao/KyberDAO.sol [hunk 6]
```diff
@@ -655,8 +621,8 @@ contract KyberDAO is IKyberDAO, EpochUtils, ReentrancyGuard, CampPermissionGroup
     }
 
     /**
-    * @dev helper func to get encoded reward and rebate
-    * @dev revert if validation failed
+    * @dev  helper func to get encoded reward and rebate
+    *       revert if validation failed
```
Comment (ilanDoron) [inline contractsSol5/Dao/KyberDAO.sol:625]: is function still required?

Comment (manhlx3006) [reply contractsSol5/Dao/KyberDAO.sol:625]: it is used to get encoded data from reward & rebate, useful for creating new BRR campaign

File: contractsSol5/Dao/mock/MockMaliciousDAO.sol (hunk 2 above)
Comment (manhlx3006) [reply contractsSol5/Dao/mock/MockMaliciousDAO.sol]: Updated

File: contractsSol5/Dao/KyberDAO.sol (hunk 1 above)
Comment (manhlx3006) [reply contractsSol5/Dao/KyberDAO.sol]: Fixed

File: contractsSol5/IKyberDAO.sol (hunk 3 above)
Comment (manhlx3006) [reply contractsSol5/IKyberDAO.sol]: Fixed

File: contractsSol5/Dao/KyberDAO.sol (hunk 5 above)
Comment (manhlx3006) [reply contractsSol5/Dao/KyberDAO.sol]: Updated

**Pull request merged**